### PR TITLE
FreeIPA: Adds ipa_pwpolicy module

### LINKED ISSUE #65319: FreeIPA: There is no way to modify Password Policies
##### SUMMARY

There is no way to modify Password Policies; I will be contributing a module in an upcoming PR.
##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

There is no module for modifying Password Policies.

##### ADDITIONAL INFORMATION

```
$ ansible --version
ansible 2.9.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/b1czu/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.8/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.8.0 (default, Oct 23 2019, 18:51:26) [GCC 9.2.0]

```

### PR DESCRIPTION
**Pull request by b1czu**
##### SUMMARY

This PR adds an ipa_pwpolicy module which allows to manage IPA Password Policies.

Fixes #65319 
##### ISSUE TYPE

- New Module Pull Request

##### ADDITIONAL INFORMATION

We did not have a ipa_pwpolicy module, so here's one! Borrowed a lot from @fxfitz's work (thanks! made it much easier).

### CODE REVIEW

**Comment by b1czu** [comment]
ready_for_review

File: lib/ansible/modules/identity/ipa/ipa_pwpolicy.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: ipa_pwpolicy
+author: Damian Bicz (@b1czu)
+short_description: Manage FreeIPA Password Policies
+description:
+- Add, delete and modify an IPA Password Policies using IPA API. Ommited values are not changed during module execution.
```
Comment (Akasurde) [inline lib/ansible/modules/identity/ipa/ipa_pwpolicy.py]:
```suggestion
- Add, delete and modify an IPA Password Policies using IPA API.
- Omitted values are not changed during module execution.
```

File: lib/ansible/modules/identity/ipa/ipa_pwpolicy.py [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+options:
+  cn:
+    description: Policy name.
+    required: true
+    type: str
+    aliases: ["group"]
+  cospriority:
+    description: Priority of the policy (higher number means lower priority). Ignored if group=global_policy
```
Comment (Akasurde) [inline lib/ansible/modules/identity/ipa/ipa_pwpolicy.py]:
```suggestion
    description:
    - The priority of the policy (higher number means lower priority).
    - Ignored if C(group=global_policy).
```

File: lib/ansible/modules/identity/ipa/ipa_pwpolicy.py [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+    required: true
+    type: int
+    aliases: ["priority"]
+  state:
+    description: State to ensure
```
Comment (Akasurde) [inline lib/ansible/modules/identity/ipa/ipa_pwpolicy.py]:
```suggestion
    description: State to ensure.
```

File: lib/ansible/modules/identity/ipa/ipa_pwpolicy.py [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+    ipa_host: localhost
+    ipa_user: admin
+    ipa_pass: supersecret
+'''
+
+RETURN = '''
+pwpolicy:
+  description: Password policy as returned by IPA API
+  returned: always
+  type: dict
```
Comment (Akasurde) [inline lib/ansible/modules/identity/ipa/ipa_pwpolicy.py:128]: add a sample here - 

```
sample: {
	...
}
``

**Review by Akasurde — CHANGES_REQUESTED** [review]
Rest LGTM

**Comment by Akasurde** [comment]
> ```
> lib/ansible/modules/identity/ipa/ipa_pwpolicy.py:0:0: doc-required-mismatch: Argument 'ipa_pass' in argument_spec is not required, but is documented as being required
> ```

Fixed in https://github.com/ansible/ansible/pull/66194

**Comment by b1czu** [comment]
ready_for_review

**Comment by b1czu** [comment]
bot_status

**Pull request closed without merging**